### PR TITLE
Filesystem.source()

### PR DESCRIPTION
**Pull request by swankjesse**

### CODE REVIEW

**Review by JakeWharton — APPROVED** [review]

File: okio-files/src/commonTest/kotlin/okio/files/FileSystemTest.kt [hunk 1]
```diff
@@ -45,4 +46,26 @@ class FileSystemTest {
     } catch (expected: IOException) {
     }
   }
+
+  @Test
+  fun `file source no such directory`() {
+    try {
```
Comment (Egorand) [inline okio-files/src/commonTest/kotlin/okio/files/FileSystemTest.kt:52]: nit: `assertThrows`?

File: okio-files/src/macosX64Main/kotlin/okio/FileSource.kt [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+import platform.posix.feof
+import platform.posix.ferror
+import platform.posix.fread
+import platform.posix.free
+
+/** Reads the bytes of a file as a source. */
+internal class FileSource (
+  private val file: CPointer<FILE>
+) : Source {
+  private var segmentSize = 8192L
```
Comment (Egorand) [inline okio-files/src/macosX64Main/kotlin/okio/FileSource.kt:34]: nit: companion object or top-level `const`

**Review by Egorand — APPROVED** [review]

File: okio-files/src/commonTest/kotlin/okio/files/FileSystemTest.kt (hunk 1 above)
Comment (swankjesse) [reply okio-files/src/commonTest/kotlin/okio/files/FileSystemTest.kt:52]: We don’t have that in Okio?

File: okio-files/src/macosX64Main/kotlin/okio/FileSource.kt (hunk 2 above)
Comment (swankjesse) [reply okio-files/src/macosX64Main/kotlin/okio/FileSource.kt]: Fixed!

**Pull request merged**